CRM/Case/BAO/Case.php

```diff
@@ -2995,11 +2995,28 @@ public static function buildOptions($fieldName, $context = NULL, $props = []) {
 
       // Filter status id by case type id
       case 'status_id':
-        if (!empty($props['case_type_id']) && is_scalar($props['case_type_id'])) {
-          $idField = is_numeric($props['case_type_id']) ? 'id' : 'name';
-          $caseType = civicrm_api3('CaseType', 'getsingle', [$idField => $props['case_type_id'], 'return' => 'definition']);
-          if (!empty($caseType['definition']['statuses'])) {
-            $params['condition'] = 'v.name IN ("' . implode('","', $caseType['definition']['statuses']) . '")';
+        if (!empty($props['case_type_id'])) {
+          // cast single values to a single value array
+          $caseTypeIdValues = is_array($props['case_type_id']) ? $props['case_type_id'] : [$props['case_type_id']];
+
+          $idField = is_numeric($caseTypeIdValues[0]) ? 'id' : 'name';
+          $caseTypeDefs = (array) \Civi\Api4\CaseType::get(TRUE)
```

Comment:
This flips the permission switch (the old code used api3 which defaults to FALSE.
I'm not exactly sure which is more correct but would recommend keeping it consistent so changing this to
```suggestion
          $caseTypeDefs = (array) \Civi\Api4\CaseType::get(FALSE)
```